src/ncp/ncp_base_mtd.cpp

```diff
@@ -1013,6 +1013,38 @@ template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_THREAD_PENDING_DATASE
     return EncodeOperationalDataset(dataset);
 }
 
+#if OPENTHREAD_ENABLE_JOINER
+template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_MESHCOP_JOINER_STATE>(void)
```

Comment:
Is it better to send notification of this state instead of provide a getter? We could remove memory used by joiner after thread started in future.